### PR TITLE
Refractoring

### PR DESCRIPTION
**Pull request by Frityet**

### CODE REVIEW

File: QuantumLink/Common/Networking/Protocol/Packet.cs [hunk 1]
```diff
@@ -2,50 +2,50 @@
...
         {
             this.Opcode = opcode;
-            this.stream = stream;
+            this.Stream = stream;
         }
     }
 
     public class OutboundPacket : Packet
     {
-        protected readonly BinaryWriter writer;
```
Comment (TheRealMichaelWang) [inline QuantumLink/Common/Networking/Protocol/Packet.cs:21]: Why use get. Readonly is more use-case specific, and accurate

File: QuantumLink/Common/Networking/Protocol/Requests/AccountRequest.cs [hunk 2]
```diff
@@ -30,8 +28,8 @@ public enum AccountOperation
             DeleteAccount
         }
 
-        public readonly AccountOperation Operation;
```
Comment (TheRealMichaelWang) [inline QuantumLink/Common/Networking/Protocol/Requests/AccountRequest.cs:33]: Again, I'm not sure why readonly cannot be used. See this stack overflow answer: https://stackoverflow.com/questions/37496738/c-sharp-readonly-vs-get. By using readonly, we define these properties as feilds intead, which are much faster than going around a get and set.

File: QuantumLink/Common/Networking/Protocol/Requests/AuthRequest.cs [hunk 3]
```diff
@@ -1,33 +1,30 @@
...
             }
         }
 
-        public readonly string Username;
-        public readonly string Password;
-
-        public readonly bool CreateAccount;
+        public string Username      { get; }
+        public string Password      { get; }
+        public bool   CreateAccount { get; }
```
Comment (TheRealMichaelWang) [inline QuantumLink/Common/Networking/Protocol/Requests/AuthRequest.cs]: Why the spacing in between types and identifiers

File: QuantumLink/Server/Networking/Session.cs [hunk 4]
```diff
@@ -1,65 +1,65 @@
...
 
-        protected readonly TcpClient client;
-        protected readonly NetworkStream networkStream;
+        protected readonly TcpClient Client;
+        protected readonly NetworkStream NetworkStream;
 
-        protected readonly Thread listenThread;
-        private volatile bool stopping;
+        protected readonly Thread ListenThread;
+        private volatile bool _stopping;
```
Comment (TheRealMichaelWang) [inline QuantumLink/Server/Networking/Session.cs:20]: Whats up with the naming conventions. I think we should establish a unified naming convention for this project before moving onwards

**Review by TheRealMichaelWang — COMMENTED** [review]
Most of your changes are great, especially the namespace reorganizing, but I am still quite adamant that ```readonly``` be used instead of ```{get;}``` for the following reasons:
- Exposing the getter says nothing about *whether* a value is read-only or not
- Using a ```readonly``` field is faster since properties are really syntactic sugar for get/set over a field, 
- The readonly modifier explicitly enforces the immutability of a value after assignment.
That brings me to my second point, about defining ```{get; set;}```'s for existing properties with access modifiers. Get and set are implicitly defined, theres no point in explicitly typing it out.

File: QuantumLink/Server/Networking/Session.cs (hunk 4 above)
Comment (Frityet) [reply QuantumLink/Server/Networking/Session.cs:20]: Yes, the microsoft one

File: QuantumLink/Common/Networking/Protocol/Packet.cs (hunk 1 above)
Comment (Frityet) [reply QuantumLink/Common/Networking/Protocol/Packet.cs:21]: Properties are better

File: QuantumLink/Common/Networking/Protocol/Requests/AccountRequest.cs (hunk 2 above)
Comment (Frityet) [reply QuantumLink/Common/Networking/Protocol/Requests/AccountRequest.cs:33]: Properties are better

File: QuantumLink/Common/Networking/Protocol/Requests/AuthRequest.cs (hunk 3 above)
Comment (Frityet) [reply QuantumLink/Common/Networking/Protocol/Requests/AuthRequest.cs]: IDE setting I left on, my apologies

**Comment by Frityet** [comment]
Properties are better

**Comment by Frityet** [comment]
> Exposing the getter says nothing about whether a value is read-only or not
with a getter, you are saying that this value can only be get, and no matter what, cannot be set

> Using a readonly field is faster since properties are really syntactic sugar for get/set over a field,

It is the same amount of time, and get;set; is the standard

> The readonly modifier explicitly enforces the immutability of a value after assignment.
That brings me to my second point, about defining {get; set;}'s for existing properties with access modifiers. Get and set are implicitly defined, theres no point in explicitly typing it out.

Whilst it is true that in the end, the compiler gens the same code, that does not dismiss the fact that properties are the standard for public variables in classes. It is better to stick to the standard than to go through with something that seems a bit more convinent

**Comment by TheRealMichaelWang** [comment]
Can you at least resolve the spacing issues before I merege?

**Comment by Frityet** [comment]
alright

**Pull request merged**